src/ch/blinkenlights/android/vanilla/PlaybackService.java

```diff
@@ -1579,8 +1592,24 @@ public boolean handleMessage(Message message)
 				ArrayList<Long> items = mPlayCounts.getTopSongs(mAutoPlPlaycounts);
 				Playlist.addToPlaylist(resolver, id, items);
 			}
+			break;
+		case MSG_HANDLE_PLUGIN:
+			// reset plugin collection timer
+			mHandler.removeMessages(MSG_DELAY_PLUGIN_COLLECTION);
 
+			Intent descIntent = (Intent) message.obj;
+			String pluginName = descIntent.getStringExtra(PluginUtils.EXTRA_PARAM_PLUGIN_NAME);
+			ApplicationInfo info = descIntent.getParcelableExtra(PluginUtils.EXTRA_PARAM_PLUGIN_APP);
+			mPlugins.put(pluginName, info);
 
+			mHandler.sendEmptyMessageDelayed(MSG_DELAY_PLUGIN_COLLECTION, 200);
```

Comment:
Where does the 200ms come from?
Wouldn't it be possible to do this without a magic timeout? (eg: query the package manager how many plugins are actually installed and check if all of them replied?)